cores/esp8266/WString.h

```diff
@@ -288,19 +292,24 @@ class String {
         enum { SSOSIZE = sizeof(struct _ptr) + 4 - 1 }; // Characters to allocate space for SSO, must be 12 or more
         struct _sso {
             char     buff[SSOSIZE];
-            unsigned char len   : 7; // Ensure only one byte is allocated by GCC for the bitfields
-            unsigned char isSSO : 1;
+            unsigned char len    : 7; // Ensure only one byte is allocated by GCC for the bitfields
+            unsigned char isHeap : 1;
         } __attribute__((packed)); // Ensure that GCC doesn't expand the flag byte to a 32-bit word for alignment issues
         enum { CAPACITY_MAX = 65535 }; // If typeof(cap) changed from uint16_t, be sure to update this enum to the max value storable in the type
         union {
+            struct {
+                char _offset0 = 0; // wbuffer()[0] = 0;
+                char _offset1[SSOSIZE - 1];
+                char _offset11 = 0; // setSSO(true); setLen(0);
```

Comment:
I get why you want to do this, but I think the naming is a bit strange here.
Also, why not just use a constructor in one (not both) of the existing structs `_ptr` or `_sso` ?

I think `_sso` is the best choice here for having the constructor to zero all values.
And since it has to be done in the body of the constructor anyway, you can also make a function in that struct and call it both from its constructor and from the `String::init()` function.

As done here, using an unnamed struct with `_offsetNN` names is confusing.